user_sync/rules.py

```diff
@@ -406,12 +408,18 @@ def read_desired_user_groups(self, mappings, directory_connector_manager):
                 target_attributes['lastname'] = directory_user.get('lastname')
                 target_attributes['country'] = directory_user.get('country')
                 self.after_mapping_hook_scope['target_attributes'] = target_attributes
-
                 # invoke the customer's hook code
+                self.after_mapping_hook_scope['drop'] = self.drop_user()
```

Comment:
I think we'll want to do this slightly different.  Don't call drop_user here - just pass the reference so it can be called like drop() in the extension